client/src/pages/Availability/Availability.tsx

```diff
@@ -0,0 +1,222 @@
+import { Button, InputLabel, FormControl } from '@mui/material';
+import { Box } from '@mui/system';
+import SettingHeader from '../../components/SettingsHeader/SettingsHeader';
+import { Typography } from '@mui/material';
+import Select from '@mui/material/Select';
+import MenuItem from '@mui/material/MenuItem';
+import createAvailbilityRow from './AvailInterface';
+import { Formik, FormikHelpers } from 'formik';
+import React, { useState, useEffect } from 'react';
+import FormInput from '../../components/FormInput/FormInput';
+import createSchedule from '../../helpers/APICalls/createSchedule';
+import { useAuth } from '../../context/useAuthContext';
+import { useSnackBar } from '../../context/useSnackbarContext';
+import getSchedule from '../../helpers/APICalls/getSchedules';
+import getSchedulesById from '../../helpers/APICalls/getScheduleById';
+import { AvailabileValue } from '../../interface/AvailabilityApiData';
+import Record from './Records';
+import useStyles from './useStyle';
+const week = ['monday', 'tuesday', 'wednesday', 'thursday', 'friday', 'saturday', 'sunday'];
+
+export default function Availability(): JSX.Element {
+  const classes = useStyles();
+  const [schedules, setSchedules] = useState([{ name: '', id: 'Select' }]);
+  const [newSchedule, setNewSchedule] = useState(false);
+  const [daysInfo, setDaysInfo] = useState<any>([]);
+  const { updateLoginContext } = useAuth();
+  const { updateSnackBarMessage } = useSnackBar();
+  const [selectedSchedule, setSelectedSchedule] = useState('');
+  const [check, setCheck] = useState(1);
+  const [dropDownSelected, setDropDownSelected] = useState(false);
+  const [available, setAvailable] = useState<AvailabileValue>();
+  const [checkExistSchedule, setCheckExistSchedule] = useState(false);
+
+  const initialData = {
+    active: false,
+    startTime: '9:00',
+    endTime: '5:00',
+  };
+  const generateSchedules = (scheduleName: any, id: any) => {
+    return (
+      <MenuItem
+        onClick={(e) => {
```

Comment:
It is not recommended to use inline function defs as it may impact performance in React. Also, we want to be able to handle promise rejection in case we weren't able to fetch the selected schedule with its id :)